src/main/java/com/vmware/vipclient/i18n/messages/service/LocaleService.java

```diff
@@ -4,179 +4,125 @@
-            Map<String, String> regionMap = null;
-            logger.trace("look for region list of '" + language + "' from cache");
-            Cache c = VIPCfg.getInstance().getCacheManager()
-                    .getCache(VIPCfg.CACHE_L2);
-            if (c != null) {
-            	FormatCacheItem cacheItem = (FormatCacheItem) c.get(REGION_PREFIX
-                        + language);    
-                regionMap = cacheItem == null ? regionMap : cacheItem.getCachedData();
-            }
-            if (regionMap != null) {
-                respMap.put(language, regionMap);
-                continue;
-            }
-            logger.trace("get region list of '" + language
-                    + "' data from backend");
-            Map<String, String> tmpMap = new RemoteLocaleOpt()
-				        .getTerritoriesFromCLDR(language);
-            regionMap = JSONUtils.map2SortMap(tmpMap);
-            respMap.put(language, regionMap);
-            if (c != null) {
-            	FormatCacheItem cacheItem = new FormatCacheItem(regionMap);
-                c.put(REGION_PREFIX + language, cacheItem);
-            }
-        }
-        return respMap;
-    }
 
-    public Map<String, String> getDisplayNamesFromCLDR(String language, 
-    		ListIterator<DataSourceEnum> msgSourceQueueIter) {
-    	Map<String, String> dispMap = new HashMap<String, String>(); 	
-    	if (!msgSourceQueueIter.hasNext()) 
-    		return dispMap;
-        
-        logger.trace("look for displayNames from cache");
-        Cache c = VIPCfg.getInstance().getCacheManager()
-                .getCache(VIPCfg.CACHE_L2);
-        if (c != null) {
-        	FormatCacheItem cacheItem = (FormatCacheItem) c.get(DISPN_PREFIX + language); 
-        	if (cacheItem == null) {
-        		cacheItem = new FormatCacheItem();
-        	}
-            dispMap = cacheItem.getCachedData();
-            if (dispMap.isEmpty()) {
-            	DataSourceEnum dataSource = (DataSourceEnum) msgSourceQueueIter.next();
-            	Map<String, String> tmpMap = dataSource.createLocaleOpt().getSupportedLanguages(language);
-                dispMap = JSONUtils.map2SortMap(tmpMap);
-                if (dispMap != null && dispMap.size() > 0) {
-                    c.put(DISPN_PREFIX + language, new FormatCacheItem(dispMap));
-                }
-            }
-        }
-        
+    private Map<String, String> getSupportedLanguagesFromDS(String locale, ListIterator<DataSourceEnum> msgSourceQueueIter) {
+        Map<String, String> dispMap = new HashMap<String, String>();
+        if (!msgSourceQueueIter.hasNext())
+            return dispMap;
+        DataSourceEnum dataSource = (DataSourceEnum) msgSourceQueueIter.next();
+        dispMap = dataSource.createLocaleOpt(dto).getSupportedLanguages(locale);
         if (dispMap == null || dispMap.isEmpty()) {
-        	return getDisplayNamesFromCLDR(language, msgSourceQueueIter);
+            dispMap = getSupportedLanguagesFromDS(locale, msgSourceQueueIter);
```

Comment:
Actually I resolve this in cache update commit, since I plan to put cache update in a new PR, so I reset that commit, after reset the change reset, too. So I resolve this in this PR again.